### PR TITLE
[ui_search] Update the content-view entity for ui-search.

### PR DESCRIPTION
**Pull request by kbidarkar**
I have ran the search tests and it has passed.

For
a) test_cv_create
b) test_cv_create_negative

```
(robottelo)[kbidarka@xyz robottelo]$ nosetests -m test_cv_create tests.foreman.ui.test_contentviews
..
----------------------------------------------------------------------
Ran 2 tests in 352.903s

OK
```

### CODE REVIEW

**Comment by kbidarkar** [comment]
Related to issue #2913

**Comment by oshtaier** [comment]
Please, run also `test_cv_delete` test

**Comment by kbidarkar** [comment]
Even these tests pass.
```
(robottelo)[kbidarka@xyz robottelo]$ nosetests -m test_cv_delete tests.foreman.ui.test_contentviews
.
----------------------------------------------------------------------
Ran 1 test in 376.079s

OK
```

**Comment by kbidarkar** [comment]
The results:

```
(robottelo)[kbidarka@dhcp201-208 robottelo]$ nosetests -m test_cv_create tests.foreman.ui.test_contentviews
..
----------------------------------------------------------------------
Ran 2 tests in 401.480s

OK
(robottelo)[kbidarka@dhcp201-208 robottelo]$ nosetests -m test_create_domain_2 tests.foreman.ui.test_domain
.
----------------------------------------------------------------------
Ran 1 test in 83.918s

OK
(robottelo)[kbidarka@dhcp201-208 robottelo]$ nosetests -m test_create_domain_1 tests.foreman.ui.test_domain
.
----------------------------------------------------------------------
Ran 1 test in 98.747s

OK
(robottelo)[kbidarka@dhcp201-208 robottelo]$ nosetests -m test_positive_create_hostgroup tests.foreman.ui.test_hostgroup
.
----------------------------------------------------------------------
Ran 1 test in 113.067s

OK
(robottelo)[kbidarka@dhcp201-208 robottelo]$ nosetests -m test_positive_create_hostgroup tests.foreman.ui.test_hostgroup
.
----------------------------------------------------------------------
Ran 1 test in 108.730s

OK
(robottelo)[kbidarka@dhcp201-208 robottelo]$ nosetests -m test_positive_create_hostgroup tests.foreman.ui.test_hostgroup
.
----------------------------------------------------------------------
Ran 1 test in 117.278s

OK
```

**Comment by oshtaier** [comment]
hold on a little on development once we add few updates for generic template as per today discussion with @elyezer

File: robottelo/ui/base.py [hunk 1]
```diff
@@ -94,10 +94,15 @@ def search(self, element_name, element_locator, search_key=None,
...
+        if katello or len(element_name) <= 30:
+            strategy, value = element_locator
+        elif not katello and len(element_name) > 30:
+            strategy, value = common_locators['select_filtered_entity']
         element = self.wait_until_element(
-            (element_locator[0], element_locator[1] % element_name),
+            (strategy, value % element_name),
             timeout=result_timeout,
         )
+
```
Comment (sthirugn) [inline robottelo/ui/base.py]: Can this be simplified by calling the following?
```py
strategy, value = element_locator
if not katello and len(element_name) > 30:
    strategy, value = common_locators['select_filtered_entity']
```

File: robottelo/ui/contentviews.py [hunk 2]
```diff
@@ -95,17 +95,18 @@ def delete_version(self, name, version):
...
-        return element
+        Navigator(self.browser).go_to_content_views()
+        return super(ContentViews, self).search(
+            element_name,
+            locators['contentviews.key_name'] or element_locator,
+            search_key=search_key,
+            katello=katello,
+            button_timeout=button_timeout,
+            result_timeout=result_timeout,
+        )
```
Comment (sthirugn) [inline robottelo/ui/contentviews.py]: May be we should standardize the `element_name` parameter across different search() definitions of entities.

Comment (sthirugn) [inline robottelo/ui/contentviews.py]: Can we have a default 'search_key' in case the caller did not specify it.

File: robottelo/ui/hostgroup.py [hunk 3]
```diff
@@ -27,10 +27,18 @@ def create(self, name, parent=None, environment=None, content_source=None,
...
-    def search(self, name):
+    def search(self, name, element_locator=None, search_key=None,
+               katello=False, button_timeout=15, result_timeout=15):
         """Searches existing hostgroup from UI."""
         Navigator(self.browser).go_to_host_groups()
-        return self.search_entity(name, locators['hostgroups.hostgroup'])
+        return super(Hostgroup, self).search(
+            name,
+            locators['hostgroups.hostgroup'] or element_locator,
+            search_key=search_key,
```
Comment (sthirugn) [inline robottelo/ui/hostgroup.py]: Can we have a default 'search_key' in case the caller did not specify it.

**Comment by sthirugn** [comment]
also +1 to @oshtaier's comment.

File: robottelo/ui/contentviews.py (hunk 2 above)
Comment (kbidarkar) [inline robottelo/ui/contentviews.py]: I believe 'search_key' is only required for the user entity which would be "login". For others it's not required by default.

File: robottelo/ui/base.py (hunk 1 above)
Comment (oshtaier) [inline robottelo/ui/base.py]: Sounds really great

@kbidarkar Please, submit that important fix into new branch!

File: robottelo/ui/contentviews.py (hunk 2 above)
Comment (sthirugn) [inline robottelo/ui/contentviews.py]: @kbidarkar can you confirm that please

Comment (sthirugn) [inline robottelo/ui/contentviews.py]: and probably add a python comment somewhere in the code for reference.

Comment (oshtaier) [inline robottelo/ui/contentviews.py]: anyway, that is not actual anymore as we are removing search methods from the code

**Comment by oshtaier** [comment]
Add
```
strategy, value = element_locator
if not katello and len(element_name) > 30:
    strategy, value = common_locators['select_filtered_entity']
```
to new search methods when https://github.com/SatelliteQE/robottelo/pull/2984 is submitted and close current PR

NACK

**Pull request closed without merging**